tools/check-symbol-names

```diff
@@ -20,11 +20,19 @@ set -o pipefail
 
 nm -A -P "$@" | awk '
 # ignore caml_foo, camlFoo_bar, _caml_foo, _camlFoo_bar
-$2 ~ /^(_?caml[_A-Z])/ { next }
+$2 ~ /^(_?[c|C]aml[_A-Z])/ { next }
 # ignore local and undefined symbols
-$3 ~ /^[rbdtsU]$/ { next }
+$3 ~ /^[a-zU]$/ { next }
```

Comment:
Character ranges and case are weird because of posix and locales. (See [gawk docs for a lament](https://www.gnu.org/software/gawk/manual/html_node/Ranges-and-Locales.html)). Changing `[rbdtsU]` to `[a-zU]` sounds good, but to make it reliable we need to change `awk` to `LC_ALL=C awk`.

Reply:
Indeed - see also #8986!